app/models/container_template.rb

```diff
@@ -0,0 +1,55 @@
+class ContainerTemplate < ApplicationRecord
+  include CustomAttributeMixin
+
+  belongs_to :ext_management_system, :foreign_key => "ems_id"
+  belongs_to :container_project
+  has_many :container_template_parameters, :dependent => :destroy
+  has_many :labels, -> { where(:section => "labels") },
+           :class_name => CustomAttribute,
+           :as         => :resource,
+           :dependent  => :destroy
+
+  serialize :objects, Array
+
+  acts_as_miq_taggable
+
+  def instantiate(params)
+    temp = {}
+    temp[:metadata] = {}
+    temp[:metadata][:name] = name
+    temp[:metadata][:namespace] = container_project.name
+    temp[:objects] = objects
+    temp[:parameters] = params
+
+    ext_management_system.with_provider_connection do |client|
+      kubeclient = Marshal.load(Marshal.dump(client))
+      kubeclient.api_endpoint.path = "/api"
```

Comment:
This is somewhat hacky since it has assumptions on kubeclient internal state.
you can  use the ext_management_system method that does not use a block to get two clients and do the clean up yourself.

also see [this](https://github.com/ManageIQ/manageiq/blob/e079737fbc5b23001b36091bc9db3aedeea5c647/app/models/manageiq/providers/openshift/container_manager/refresher.rb#L15)

In the future it would be helpful if we could provide a forwarding client, just like the oc client lets us create objects in OpenShift and k8s alike cc @abonas @simon3z.